### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by louismun**
Refactored existing code to reduce repetition & ambiguity
@StevenPurn @swiimp

### CODE REVIEW

**Comment by FermiDirak** [comment]
no bueno
👎

**Review by swiimp — APPROVED** [review]
Looks fine

File: client/src/components/OverallRatings.jsx [hunk 1]
```diff
@@ -18,19 +18,17 @@ class OverallRatings extends React.Component {
             starCount = { 1: 0, 2: 0, 3: 0, 4: 0, 5: 0 },
             bar5length = 0, bar4length = 0, bar3length = 0, bar2length = 0, bar1length = 0;
 
-        for (var i = 0; i < this.props.reviewData.length; i++) {
+        for (let i = 0; i < this.props.reviewData.length; i++) {
```
Comment (beth) [inline client/src/components/OverallRatings.jsx:21]: Problem(s): I find this part of the render method to be difficult to follow and add to a feeling of clutter/disorganization. It's also impossible to test this part separately from the component's render.

Fix: All of this logic should be refactored into a function(/s). You could put them a separate file OR use named exports in this file for testing purposes.

Comment (beth) [reply client/src/components/OverallRatings.jsx:21]: (I can't attach a comment to the specific lines)
Problem: The hardcoded URLs (lines 11-14, 73-81) cause this component to be difficult to reuse.

Fix: Separate file for static asset image config, imported into this component and referenced.

File: client/src/components/OverallRatings.jsx [hunk 2]
```diff
@@ -51,21 +49,18 @@ class OverallRatings extends React.Component {
...
-                        <div className='overall-rating-mini'><div className='bar-title'>3 STAR</div> <div style={{ width: bar3length + 'px' }} className='bar-left'></div><div style={{ width: 200 - bar3length + 'px' }} className='bar-right'></div><div className='bar-count'>{starCount[3]}</div></div>
-                        <div className='overall-rating-mini'><div className='bar-title'>2 STAR</div> <div style={{ width: bar2length + 'px' }} className='bar-left'></div><div style={{ width: 200 - bar2length + 'px' }} className='bar-right'></div><div className='bar-count'>{starCount[2]}</div></div>
-                        <div className='overall-rating-mini'><div className='bar-title'>1 STAR</div> <div style={{ width: bar1length + 'px' }} className='bar-left'></div><div style={{ width: 200 - bar1length + 'px' }} className='bar-right'></div><div className='bar-count'>{starCount[1]}</div></div>
+                        {barlength.map((item, index) =>
```
Comment (beth) [inline client/src/components/OverallRatings.jsx:52]: The map is much better. 

Problem: The use of "magic numbers" (ie, the 200, the 5) make it difficult to know what to change and how if something needs to be adjusted. 

Fix: You can declare the numbers as variables at the top of the file with reasonable names (all caps variable name is a common convention for this) and then use the variables in place of the numbers.

File: client/src/components/OverallRatings.jsx [hunk 3]
```diff
@@ -51,21 +49,18 @@ class OverallRatings extends React.Component {
...
                     </div>
-                        <p className='review-feedback'> Tell us what you think about this product. We value your feedback.
-                    <div className='padding'></div>
-                            <button className='review-feedback-button'>WRITE A REVIEW</button></p>
+                        <p className='review-feedback'> Tell us what you think about this product. We value your feedback.</p>
+                        <div className='padding'></div>
+                        <button className='review-feedback-button'>WRITE A REVIEW</button>
                     </div>
                 </div>
                 <div className='padding' style={{ padding: '30px' }}></div>
```
Comment (beth) [inline client/src/components/OverallRatings.jsx:66]: Problem: A mix of inline CSS/style sheets makes it hard to track down where styles are coming from. 

Fix: Use Styled Components or CSS Modules.

File: client/src/components/ReviewBody.jsx [hunk 4]
```diff
@@ -0,0 +1,22 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import axios from 'axios';
+
+let ReviewBody = (props) => {
+    let date = props.reviewData.date;
```
Comment (beth) [inline client/src/components/ReviewBody.jsx:6]: Problem: This formatting logic clutters the component and the formatting logic is impossible to test separately from the component render.

Fix: Place this logic in an appropriately named function and invoke it. Separate out into a different file or export from this file using named exports.

File: client/src/components/ReviewBody.jsx [hunk 5]
```diff
@@ -0,0 +1,22 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import axios from 'axios';
```
Comment (beth) [inline client/src/components/ReviewBody.jsx:3]: Does this component need axios and react-dom?

File: client/src/components/ReviewList.jsx [hunk 6]
```diff
@@ -15,11 +15,10 @@ class Reviews extends React.Component {
 
     componentDidMount() {
         const context = this;
-        var url = window.location.href.split('/').pop();
+        let url = window.location.href.split('/').pop();
```
Comment (beth) [inline client/src/components/ReviewList.jsx:18]: Problem: Getting the id from the url makes it difficult to reuse this component in other scenarios.

Fix: Have the ReviewList component accept the id as a prop.

File: database/generate.js [hunk 7]
```diff
@@ -1,27 +1,27 @@
...
-        var pickGender = Math.floor(Math.random() * genderArray.length);
-        var chestSize = Math.floor(Math.random() * (52 - 30 + 1)) + 30;
-        var waistSize = Math.floor(Math.random() * (35 - 28 + 1)) + 28;
-        var hipsSize = Math.floor(Math.random() * (35 - 28 + 1)) + 28;
-        var fitRating = Math.floor(Math.random() * 5) + 1;
-        var starRating = Math.floor(Math.random() * 5) + 1;
-        var lengthRating = Math.floor(Math.random() * 5) + 1;
-        var qualityRating = Math.floor(Math.random() * 5) + 1;
-        var helpfulRating = Math.floor(Math.random() * 15);
+        let pickAge = Math.floor(Math.random() * ageArray.length);
```
Comment (beth) [inline database/generate.js:13]: Problem(s): If someone wants to adjust the min/max for any of these values, it is non-obvious how to go about doing that. If you decide to use a different random number generation scheme (say you want a non-uniform distribution, for example), then there is a ton of code to change here.

Fix: Abstract this code into a `randomIntegerBetween(min, max)` function. 

Note: I also don't think saving these values into variables is helping readability. Yes, they are now named, but, I think they can be intuited from the object keys on the following lines. If you created the function above (as well as a similar function, `chooseRandomElement(array)`), then the part where you construct the object literal starting on line 27 on could look something like 

```
data.push({
  waist_size: randomIntegerBetween(28, 35),
  height: choseRandomElement(heightArray),
  // etc
})
```

**Review by beth — COMMENTED** [review]
Hi @lhmun!

Is there a reason you are committing the bundle? I'd advise against it unless you have a specific reason for doing so.

Other than that, please see comments and let me know if you have any questions!

also ccing @swiimp as the original reviewer - some of this may be interesting to you as other things to look out for while reviewing.

**Pull request merged**